payments-core/src/main/java/com/stripe/android/payments/PaymentFlowFailureMessageFactory.kt

```diff
@@ -35,7 +36,8 @@ internal class PaymentFlowFailureMessageFactory(
     private fun createForPaymentIntent(
         paymentIntent: PaymentIntent
     ) = when {
-        paymentIntent.lastPaymentError?.code == PaymentIntent.Error.CODE_AUTHENTICATION_ERROR -> {
+        (paymentIntent.status == StripeIntent.Status.RequiresAction && paymentIntent.paymentMethod?.type?.isVoucher != true) ||
```

Comment:
nit: 
```
(paymentIntent.status == StripeIntent.Status.RequiresAction && !paymentIntent.paymentMethod?.type?.isVoucher)
```

Reply:
I am having trouble with this because it could be null.